resources/type-system/inference.md

```diff
@@ -698,15 +770,32 @@ to `L` under constraints `C`.
   under constraints `Ci`.
 - Function types with named parameters are treated analogously to the positional
   parameter case above.
-- A generic function type `<T0 extends B0, ..., Tn extends Bn>F0` is a subtype
-match for a generic function type `<S0 extends B0, ..., Sn extends Bn>F1` with
-respect to `L` under constraints `Cl`:
-  - If `F0[Z0/T0, ..., Zn/Tn]` is a subtype match for `F0[Z0/S0, ..., Zn/Sn]`
-with respect to `L` under constraints `C`, where each `Zi` is a fresh type
-variable with bound `Bi`.
-  - And `Cl` is `C` with each constraint replaced with its closure with respect
+
+- A generic function type `<T0 extends B00, ..., Tn extends B0n>F0` is a subtype
+match for a generic function type `<S0 extends B10, ..., Sn extends B1n>F1` with
+respect to `L` under constraint set `C2`
+  - If `B0i` is a subtype match for `B1i` with constraint set `Ci0`
+  - And `B1i` is a subtype match for `B0i` with constraint set `Ci1`
+  - And `Ci2` is `Ci0 + Ci1`
+  - And `Z0...Zn` are fresh variables with bounds `B20, ..., B2n`
+    - Where `B2i` is `B0i[Z0/T0, ..., Zn/Tn]` if `P` is a type schema
+    - Or `B2i` is `B1i[Z0/S0, ..., Zn/Sn]` if `Q` is a type schema
+      - In other words, we choose the bounds for the fresh variables from
+        whichever of the two generic function types is a type schema and does
+        not contain any variables from `L`.
+  - And `F0[Z0/T0, ..., Zn/Tn]` is a subtype match for `F0[Z0/S0, ..., Zn/Sn]`
```

Comment:
Do you mean `F1[Z0/S1, ..., Zn/Sn]` as the second function type?

Reply:
Indeed I do.  Thanks for the close reading.